sky/shell/shell.cc

```diff
@@ -227,25 +225,82 @@ void Shell::GetPlatformViews(
   *platform_views = platform_views_;
 }
 
-void Shell::WaitForPlatformViews(
-    std::vector<base::WeakPtr<PlatformView>>* platform_views) {
+void Shell::WaitForPlatformViewIds(
+    std::vector<uintptr_t>* platform_view_ids) {
 
   base::WaitableEvent latch(false, false);
 
   ui_task_runner()->PostTask(
       FROM_HERE,
-      base::Bind(&Shell::WaitForPlatformViewsUIThread,
+      base::Bind(&Shell::WaitForPlatformViewsIdsUIThread,
                  base::Unretained(this),
-                 base::Unretained(platform_views),
+                 base::Unretained(platform_view_ids),
                  base::Unretained(&latch)));
 
   latch.Wait();
 }
 
-void Shell::WaitForPlatformViewsUIThread(
-    std::vector<base::WeakPtr<PlatformView>>* platform_views,
+void Shell::WaitForPlatformViewsIdsUIThread(
+    std::vector<uintptr_t>* platform_view_ids,
     base::WaitableEvent* latch) {
-  GetPlatformViews(platform_views);
+  std::vector<base::WeakPtr<PlatformView>> platform_views;
+  GetPlatformViews(&platform_views);
+  for (auto it = platform_views.begin(); it != platform_views.end(); it++) {
+    PlatformView* view = it->get();
+    if (!view) {
+      // Skip dead views.
+      continue;
+    }
+    platform_view_ids->push_back(reinterpret_cast<uintptr_t>(view));
+  }
+  latch->Signal();
+}
+
+void Shell::RunInPlatformView(uintptr_t view_id,
+                              const char* main_script,
+                              const char* packages_file,
+                              const char* asset_directory,
+                              bool* view_existed) {
+  base::WaitableEvent latch(false, false);
+  DCHECK(view_id != 0);
+  DCHECK(main_script);
+  DCHECK(packages_file);
+  DCHECK(asset_directory);
+  DCHECK(view_existed);
+
+  ui_task_runner()->PostTask(
+      FROM_HERE,
+      base::Bind(&Shell::RunInPlatformViewUIThread,
+                 base::Unretained(this),
+                 view_id,
+                 base::Unretained(main_script),
+                 base::Unretained(packages_file),
+                 base::Unretained(asset_directory),
+                 base::Unretained(view_existed),
+                 base::Unretained(&latch)));
```

Comment:
Shoot, I only changed the callback signature to accept std::strings I forgot to update the caller. 

This is done now.